.github/workflows/internal-archive-release.yml

```diff
@@ -250,21 +260,25 @@ jobs:
   create-internal-release:
     name: Create release for internal archive
     runs-on: ubuntu-latest
-    needs: [initial-setup, run-integration-tests-tox, run-integration-tests-hatch]
+    needs: [initial-setup, job-setup, run-integration-tests-tox, run-integration-tests-hatch]
     # Build artifact if
-    # 1. Spark invoked (it runs tests via its in-repo workflow)
-    # 2. Integration tests passed at least via one job above
+    # 1. Setup jobs succeeded
+    # 2a. Tests can be skipped
+    # 2b. One of the integration test sets passed (these only run on passing unit tests)
     if: |
-      always() && (
-        needs.initial-setup.outputs.is_spark_build == 'true' ||
-        (needs.run-integration-tests-tox.result == 'success' || needs.run-integration-tests-hatch.result == 'success')
-      )
+      always() &&
+        needs.job-setup.result == 'success' &&
+        (
+          needs.initial-setup.outputs.skip_tests == 'true' ||
+          (needs.run-integration-tests-tox.result == 'success' || needs.run-integration-tests-hatch.result == 'success')
+        )
 
     steps:
       - name: "Checkout provided ref, default to branch main"
         uses: actions/checkout@v4
         with:
-          ref: ${{ inputs.ref || 'main' }}
+          repository: "${{ inputs.org }}/dbt-${{ inputs.dbms_name }}"
+          ref: "${{ inputs.ref || 'main' }}"
```

Comment:
Could we change `ref` to be an optional input and default to `main` there?

Reply:
I used a default value to simplify this value setting. Good thing to cleanup.